water_neuron/src/conversion.rs

```diff
@@ -15,6 +19,59 @@ use icrc_ledger_types::icrc2::transfer_from::TransferFromArgs;
 pub const MINIMUM_DEPOSIT_AMOUNT: ICP = ICP::ONE;
 pub const MINIMUM_WITHDRAWAL_AMOUNT: ICP = ICP::from_unscaled(10);
 
+pub async fn cancel_withdrawal(neuron_id: NeuronId) -> Result<MergeResponse, String> {
+    let caller = ic_cdk::caller();
+    let _guard_principal =
+        GuardPrincipal::new(caller).map_err(|guard_error| match guard_error {
+            GuardError::AlreadyProcessing => "Already processing request.",
+            GuardError::TooManyConcurrentRequests => "Too many concurrent requests.",
+        })?;
+
+    let icp_stake_e8s;
+    match get_neuron_info(neuron_id.id).await? {
+        Ok(neuron_info) => icp_stake_e8s = neuron_info.stake_e8s,
+        Err(gov_err) => {
+            return Err(format!("[cancel_withdrawal] Governance error: {gov_err:?}"));
+        }
+    }
+
+    schedule_now(TaskType::ProcessLogic);
+
+    stop_dissolvement(neuron_id).await?;
+    match merge_neuron(SIX_MONTHS_NEURON_NONCE, neuron_id)
+        .await?
+        .command
+        .expect("[cancel_withdrawal] Expected a command but got None.")
+    {
+        CommandResponse::Merge(response) => {
+            schedule_now(TaskType::ProcessPendingTransfers);
```

Comment:
why? no new transfer has been scheduled.

Reply:
Shouldn't the user retrieve its NICP ? I was inspired by the ICP deposit endpoint, once the merge is done shouldn't we process an nICP transfer?

Reply:
You're right, a transfer has been queued.